File: aoc2203.py
with open('2203.0') as file:
# with open('test') as file:
    a = [_.strip() for _ in file]

def calc(c: chr) -> int:
    cc = ord(c)
    if 'a'<=c<='z':
        return cc - ord('a') + 1
    else:
        return cc - ord('A') + 27

lena = len(a)
res = 0
res2 = 0
for i in range(lena):
    s = a[i]
    n = len(s)
    l = s[:n // 2]
    r = s[n // 2:]
    for c in l:
        if c in r:
            res += calc(c)
            break
    if i > lena - 3 or i % 3 != 0:
        continue
    d = a[i + 1] # down
    dd = a[i + 2]
    ok = False
    ok2 = False
    for c in s:
        if ok and ok2:
            break
        ok = False
        ok2 = False
        if c in d:
            ok = True
        if c in dd:
            ok2 = True
        if ok and ok2:
            res2 += calc(c)
            # print(c)

print("Star 1:", res)
print("Star 2:", res2)

File: aoc2203.go
package main

import (
	"fmt"
	"bufio"
	"os"
	"strings" // strings.Contains(hays, "need")
	_ "io/ioutil"
	_ "strconv"
)

func solve(path string) (int, int) {
	data, err := os.Open(path)
	if err != nil {
		panic(err)
	}
	var a []string
	scanner := bufio.NewScanner(data)
	scanner.Split(bufio.ScanLines)
	for scanner.Scan() {
		a = append(a, scanner.Text())
	}
	res, res2 := 0, 0
	for i, line := range a {
		n := len(line)
		L := line[: n / 2]
		R := line[n / 2:]
		for _, c := range L {
			if strings.Contains(R, string(c)) {
				res += calc(c)
				break
			}
		}

		// part 2

		if i > len(a) - 3 || i % 3 != 0 {
			continue
		}
		d := a[i + 1]
		dd := a[i + 2]
		var ok, ok2 bool
		for _, c := range line {
			ok = strings.Contains(d, string(c))
			ok2 = strings.Contains(dd, string(c))
			if ok && ok2 {
				res2 += calc(c)
				break
			}
		}
	}
	return res, res2
}

func calc(c rune) int {
	cc := int(c)
	if 'a' <= c && c <= 'z' {
		return cc - int('a') + 1
	}
	return cc - int('A') + 27
}

func main() {
	var r1, r2 int

	r1, r2 = solve("_inputs/2203.0")
	fmt.Println("data")
	fmt.Println("Star 1:", r1)
	fmt.Println("Star 2:", r2, "\n")

	r1, r2 = solve("_inputs/2203.1")
	fmt.Println("test")
	fmt.Println("Star 1:", r1)
	fmt.Println("Star 2:", r2)
}
